src/explore-education-statistics-admin/src/pages/release/data/components/LiveApiDataSetsTable.tsx

```diff
@@ -74,44 +76,45 @@ export default function LiveApiDataSetsTable({
                     releaseApiDataSetDetailsRoute.path,
                     {
                       publicationId,
-                      releaseId,
+                      releaseId: releaseVersionId,
                       dataSetId: dataSet.id,
                     },
                   )}
                 >
                   View details
                   <VisuallyHidden> for {dataSet.title}</VisuallyHidden>
                 </Link>
-                {canUpdateRelease && (
-                  <ApiDataSetCreateModal
-                    buttonText={
-                      <>
-                        Create new version
-                        <VisuallyHidden> for {dataSet.title}</VisuallyHidden>
-                      </>
-                    }
-                    publicationId={publicationId}
-                    releaseId={releaseId}
-                    submitText="Confirm new data set version"
-                    title="Create a new API data set version"
-                    onSubmit={async ({ releaseFileId }) => {
-                      await apiDataSetVersionService.createVersion({
-                        dataSetId: dataSet.id,
-                        releaseFileId,
-                      });
-                      history.push(
-                        generatePath<ReleaseDataSetRouteParams>(
-                          releaseApiDataSetDetailsRoute.path,
-                          {
-                            publicationId,
-                            releaseId,
-                            dataSetId: dataSet.id,
-                          },
-                        ),
-                      );
-                    }}
-                  />
-                )}
+                {canUpdateRelease &&
+                  !dataSet.previousReleaseIds.includes(releaseId) && (
+                    <ApiDataSetCreateModal
+                      buttonText={
+                        <>
+                          Create new version
+                          <VisuallyHidden> for {dataSet.title}</VisuallyHidden>
+                        </>
+                      }
+                      publicationId={publicationId}
+                      releaseId={releaseVersionId}
+                      submitText="Confirm new data set version"
+                      title="Create a new API data set version"
+                      onSubmit={async ({ releaseFileId }) => {
+                        await apiDataSetVersionService.createVersion({
+                          dataSetId: dataSet.id,
+                          releaseFileId,
+                        });
+                        history.push(
+                          generatePath<ReleaseDataSetRouteParams>(
+                            releaseApiDataSetDetailsRoute.path,
+                            {
+                              publicationId,
+                              releaseId: releaseVersionId,
```

Comment:
Guessing this one and another on L79 is correct and will be tidied up by the big rename!

Reply:
Yup!